openwisp_controller/config/static/config/js/widget.js

```diff
@@ -1,6 +1,69 @@
 'use strict';
 (function ($) {
     var inFullScreenMode = false,
+        prevDefaultValues = {},
+        defaultValuesUrl = window.location.origin + '/admin/config/device/get-template-default-values/',
+    removeDefaultValues = function(contextValue, defaultValues) {
+        // remove obsulate default values from contextValue
```

Comment:
obsulate > obsolete, but I would just write something like "Remove default values when a template is removed"